core/templates/dev/head/components/entity-creation-services/topic-creation.service.ts

```diff
@@ -55,26 +59,28 @@ angular.module('oppia').factory('TopicCreationService', [
           ]
         });
 
-        modalInstance.result.then(function(topicName) {
-          if (topicName === '') {
+        modalInstance.result.then(function(topic) {
+          if (topic.topicName === '') {
```

Comment:
Isn't `abbreviated_topic_name` also compulsory? If not, maybe we should have a mechanism, where if the topic name  is short, automatically copy that into abbreviated_topic_name.
@seanlip Do you have any thoughts about this?

Reply:
I think it should be compulsory. Adding an abbreviated name is not hard (unlike, say, creating a topic thumbnail image).

Reply:
Done.

Reply:
Done.